src/lambdapi.ml

```diff
@@ -290,10 +287,15 @@ let lsp_log_file : string Term.t =
 
 (** Specific to the ["decision-tree"] command. *)
 
-let qsym : string list Term.t =
+let qsym : (string list * string) Term.t =
   let doc = "Fully qualified symbol name with dot separated identifiers." in
   let i = Arg.(info [] ~docv:"MOD_PATH.SYM" ~doc) in
-  Arg.(non_empty & pos 0 (list ~sep:'.' string) [] & i)
+  let separate lst =
+    let rv = List.rev lst in
+    ((List.rev (List.tl rv)), List.hd rv)
+  in
```

Comment:
I would prefer
```ocaml
let separate l =
  match List.rev l with
  | []   -> assert false (* Unreachable. *)
  | s::l -> (List.rev l, s)
in
```
since `List.hd` and `List.tl` can raise exceptions.

Also, please use `Path.t` instead of `string list` for the type.

Reply:
And rather use:
```ocaml
let arg = Arg.(non_empty & pos 0 (list ~sep:'.' string) [] & i) in
Term.(const separate $ arg)
```

Reply:
what about the one-liner?

Reply:
Yeah that is not too bad (I was actually considering asking you to do exactly that!), but I don't like to have functions that can raise exceptions outside of an exception handler. I would much rather have the `assert false` this way if a further modification leads to the error happening we will know where it comes from.